beacon-chain/sync/rpc_beacon_blocks_by_root.go

```diff
@@ -180,39 +181,35 @@ func (s *Service) sendAndSaveBlobSidecars(ctx context.Context, request types.Blo
 		log.WithFields(blobFields(sidecar)).Debug("Received blob sidecar RPC")
 	}
 
-	return s.cfg.beaconDB.SaveBlobSidecar(ctx, sidecars)
+	vscs, err := verification.BlobSidecarSliceNoop(sidecars)
+	if err != nil {
+		return err
+	}
+	for i := range vscs {
+		if err := s.cfg.blobStorage.Save(vscs[i]); err != nil {
+			return err
+		}
+	}
+	return nil
 }
 
 // constructPendingBlobsRequest creates a request for BlobSidecars by root, considering blobs already in DB.
 func (s *Service) constructPendingBlobsRequest(ctx context.Context, blockRoot [32]byte, count int) (types.BlobSidecarsByRootReq, error) {
-	knownBlobs, err := s.cfg.beaconDB.BlobSidecarsByRoot(ctx, blockRoot)
-	if err != nil && !errors.Is(err, db.ErrNotFound) {
+	have, err := s.cfg.blobStorage.Indices(blockRoot)
+	if err != nil {
 		return nil, err
 	}
 
-	knownIndices := indexSetFromBlobs(knownBlobs)
-	requestedIndices := filterUnknownIndices(knownIndices, count, blockRoot)
-
-	return requestedIndices, nil
-}
-
-// Helper function to create a set of known indices.
-func indexSetFromBlobs(blobs []*eth.DeprecatedBlobSidecar) map[uint64]struct{} {
-	indices := make(map[uint64]struct{})
-	for _, blob := range blobs {
-		indices[blob.Index] = struct{}{}
-	}
-	return indices
+	return requestsForMissingIndices(have, count, blockRoot), nil
 }
 
 // Helper function to filter out known indices.
-func filterUnknownIndices(knownIndices map[uint64]struct{}, count int, blockRoot [32]byte) []*eth.BlobIdentifier {
+func requestsForMissingIndices(haveIndices [fieldparams.MaxBlobsPerBlock]bool, count int, blockRoot [32]byte) []*eth.BlobIdentifier {
```

Comment:
`haveIndices` isn't great, could this not just be called `indices`? It makes the variable `has` on line 198 is a bit unclear

Reply:
What do you think of this? https://github.com/prysmaticlabs/prysm/pull/13190/commits/f4af7dc5b5dee43abe878db48f0eabd369312ea6